### PR TITLE
Decorate A class

### PR DESCRIPTION
**Pull request by bigmosi**
Interface
-   Created a class Nameable.
- Implemented a method called correct_name that will raise a NotImplementedError.
-  Turned Person class to Nameable
-   Made Person class inherits from Nameable
-  This class has a method correct_name implemented. where it returns the name attribute.
-   Prepare the base Decorator
-   It inherits from Nameable.
-   In the constructor assigned a nameable object from params to an instance variable.
-  Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
Prepared CapitalizeDecorator and TrimmerDecorator
-For the CapitalizeDecorator:
Created a class that inherits from the base Decorator class.
Implemented a method correct_name that capitalizes the output of @nameable.correct_name.
-For the TrimmerDecorator:
Created a class that inherits from the base Decorator class.
Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @bigmosi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no pull request summary

- [ ] Please write a meaningful title for your pull request and always add a short summary of changes introduced in the pull request. [Use this guide ](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
des2


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @bigmosi , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not follow the correct git flow, it should be pointing to develop branch, but yours is currently pointing to the main branch, kindly address that!

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class CapitalizeDecorator < Nameable
+  def initialize(nameable = nameable.new)
+    super()
+    @nameable = nameable
+  end
```
Comment (Deepakdanger) [inline capitalize_decorator.rb]: - [ ] Here `CapitalizeDecorator`should  Inherits from Decorator. && is no need to initialize again. Kindly remove the initialized part.
 ```ruby
class CapitalizeDecorator < Decorator
```

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class TrimmerDecorator < Nameable
+  def initialize(nameable = nameable.new)
+    super()
+    @nameable = nameable
+  end
```
Comment (Deepakdanger) [inline trimmer_decorator.rb]: - [ ] Here `TrimmerDecorator`should  Inherits from the Decorator. && is no need to initialize again. Kindly remove the initialized part.
 ```ruby
class TrimmerDecorator < Decorator
```

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
```
Comment (Deepakdanger) [inline trimmer_decorator.rb]: - [ ] Here slice only when greater then a certain length, Kindly use this
`@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @bigmosi, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_deorator.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable = nameable.new)
```
Comment (tresorsawasawa) [inline base_deorator.rb:4]: - [ **OPTIONAL** ] **RECOMMENDED:** Kindly make sure you update the constructor `initialize(nameable = nameable.new)` to this: `initialize(nameable)`. The current one generates an error when trying to test your app.

File: nemeable.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
```
Comment (tresorsawasawa) [inline nemeable.rb:2]: - [ **OPTIONAL** ] **RECOMMENDED:** Kindly make sure you correct the typing error of your `nemeable.rb` to `nameable.rb` as in all files you imported it you use `nameable.rb`(which is the good typing).

File: base_deorator.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
```
Comment (tresorsawasawa) [inline base_deorator.rb:3]: #- [ **OPTIONAL** ] **RECOMMENDED:** Kindly make sure you correct the typing error of your `base_deorator.rb` to `base_decorator.rb` as in all files you imported it you use `base_decorator.rb`(which is the good typing) and it generates errors when trying to test.

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @bigmosi 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**